From c355bb52188eb017d9afe966cf191e67f0db7647 Mon Sep 17 00:00:00 2001
From: AnchoWake <ren.suga@gmail.com>
Date: Mon, 30 Nov 2015 17:26:06 -0600
Subject: [PATCH] Ticket 731 Fix V2.

Fix Ticket 731
Users can now visit the information source link when clicking the post card.

Users can now visit original post url clicking the image/postcard in moderate view

Image and text unmoderated post don't show a background link tab

Removed typo error

Removed typo error
---
 public/js/widgets/posts/edit/EditablePost.js       | 26 ++++++++++++++++++++++
 .../layers/VoicePostLayersModerateAbstract.js      |  2 +-
 2 files changed, 27 insertions(+), 1 deletion(-)

diff --git a/public/js/widgets/posts/edit/EditablePost.js b/public/js/widgets/posts/edit/EditablePost.js
index fdf0819..a3803c7 100644
--- a/public/js/widgets/posts/edit/EditablePost.js
+++ b/public/js/widgets/posts/edit/EditablePost.js
@@ -33,6 +33,7 @@ Class(CV, 'EditablePost').includes(CV.WidgetUtils, CustomEventSupport, NodeSuppo
      */
     create : function create(config) {
         var type = this.prototype.format.capitalizeFirstLetter(config.sourceType);
+
         return new window.CV['EditablePost' + type](config);
     },
 
@@ -288,6 +289,31 @@ Class(CV, 'EditablePost').includes(CV.WidgetUtils, CustomEventSupport, NodeSuppo
             return this;
         },
 
+        addLinkNewtab : function addLinkNewtab() {
+            if(this.sourceType !== 'text' && this.sourceType !== 'image'){
+                var url = this.sourceUrl;
+
+                this.el.querySelector('.post-card-info').addEventListener("click", function( e ){
+                    e = window.event || e; 
+                    if(this === e.target) {
+                        var win = window.open(url, '_blank');
+                        win.focus();
+                    }
+                });
+
+                this.el.querySelector('.post-card-image-wrapper').addEventListener("click", function( e ){
+                    var svgIcon = this.querySelector('use');
+                    e = window.event || e; 
+                    if(this === e.target || svgIcon === e.target) {
+                        var win = window.open(url, '_blank');
+                        win.focus();
+                    }
+                });
+            }
+
+            return this;
+        },
+
         /* Adds the vote up/down buttons (for moderation management)
          * @method addVoteButtons <public> [Function]
          * @return EditablePost
diff --git a/public/js/widgets/voice/layers/VoicePostLayersModerateAbstract.js b/public/js/widgets/voice/layers/VoicePostLayersModerateAbstract.js
index e9d4b09..286a77b 100644
--- a/public/js/widgets/voice/layers/VoicePostLayersModerateAbstract.js
+++ b/public/js/widgets/voice/layers/VoicePostLayersModerateAbstract.js
@@ -52,7 +52,7 @@ Class(CV, 'VoicePostLayersModerateAbstract').inherits(CV.VoicePostLayers)({
             layer.addEditablePosts(postsData).getPosts().forEach(function(post) {
                 // Voice Owner / Org Member / Contributor
                 if (layers.allowPostEditing) {
-                    post.edit().addImageControls().addRemoveButton().addPublishButton();
+                    post.edit().addImageControls().addRemoveButton().addPublishButton().addLinkNewtab();
                     post.bind('dimensionsChanged', layers._reLayoutLayer);
                     return;
                 }
-- 
2.4.9 (Apple Git-60)

